src/ZEO/ClientStorage.py

```diff
@@ -504,8 +504,15 @@ def load(self, oid, version=''):
         return result[:2]
 
     def loadBefore(self, oid, tid):
+        result = self._cache.loadBefore(oid, tid)
+        if result:
+            return result
```

Comment:
So that makes it check the cache in the calling thread, just like ZEO 4 currently does. That seems safe and reasonable.